### DISCUSSION TITLE
Source code migration from github to codeberg possible?

### DISCUSSION THREAD

**Discussion by linux-lukas** [mastodon/mastodon#17086] [General]
Would there be dependencies on GitHub that would make it necessary to keep the source code on GitHub, or could it just as easily be on Codeberg?

**Comment by trwnh** [comment]
self-hosted gitea would probably be better than codeberg, but github currently provides a lot of infrastructure such as issues, discussions, ci/cd, hooks, etc. it would not be easy to migrate all of that, and certainly not "just as" easy.

it might interest you to know that there is a selfhosted gitlab at source.joinmastodon.org although it is currently down
